embassy-stm32/src/usart/buffered.rs

```diff
@@ -254,7 +254,17 @@ impl<'d, T: BasicInstance> BufferedUart<'d, T> {
     }
 
     pub fn set_config(&mut self, config: &Config) -> Result<(), ConfigError> {
-        reconfigure::<T>(config)
+        reconfigure::<T>(config)?;
+
+        T::regs().cr1().modify(|w| {
```

Comment:
could you put this reg write in `reconfigure` instead of pasting it 3 time?

Reply:
`reconfigure` is universally used for all uart flavors. For example when buffered uart is created it calls `configure` and then sets other registers that are needed for this driver:
https://github.com/embassy-rs/embassy/blob/ca27590afac33019dbc81489117b33fd7e3063a5/embassy-stm32/src/usart/buffered.rs#L233-L241

So putting it to `confgure` is not a good place. I'll check if I can refactor it a little to remove copy-pasting.

Reply:
I cleaned it up a little by removing redundant `w.set_fifoen(true); ` which is set in `configure` anyway.
There isn't a good place where to put those copy pasted lines without introducing more complexity.
`configure` configures all the basic stuff which is used in all uart flavours and then extra configuration is done in `buffered.rs` which is needed for this specific impl.

I would keep it as is without adding extra stuff to basic configuration.
Although I was thinking that maybe we can remove those `self.set_config` functions altogether and only rely on `SetConfig` trait as in [here](https://github.com/embassy-rs/embassy/blob/ca27590afac33019dbc81489117b33fd7e3063a5/embassy-stm32/src/usart/buffered.rs#L117C18-L142).

Then all the configuration related stuff is neatly together and there is less chance that somebody forgets to update something. However then again `SetConfig` trait must be imported by the user which is inconvenient.

Reply:
oh okay! I had missed that, sounds good then! thanks :)